src/test/java/com/github/tomakehurst/wiremock/client/ClientAuthenticationAcceptanceTest.java

```diff
@@ -142,6 +145,26 @@ void canRequireHttpsOnAdminApi() {
     assertThat(response.content(), containsString("HTTPS is required for accessing the admin API"));
   }
 
+  @Test
+  void responds404WhenRequireHttpsOnAdminApiAndNonAdminUrlWithNoStubMatch() {
```

Comment:
You're right - existing code seems to work the way I'd expect. I guess things have moved on a bit since 2021 😆 
Feel free to keep the test if you'd like, but the code change looks like it can be abandoned.
I've also tested against the latest standalone jar (3.0.1) and confirmed that it is working as I'd expect.